library/src/blas1/rocblas_reduction_impl.hpp

```diff
@@ -0,0 +1,202 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#pragma once
+
+#include "logging.h"
+#include "rocblas_reduction_template.hpp"
+#include "utility.h"
+
+template <bool ISBATCHED, typename Ti>
+void reduction_log_bench(rocblas_handle handle,
+                         rocblas_int    n,
+                         const Ti*      x,
+                         rocblas_int    incx,
+                         rocblas_stride stridex,
+                         rocblas_int    batch_count,
+                         const char*    name)
+{
+    if(ISBATCHED)
+    {
+        log_bench(handle,
+                  "./rocblas-bench",
+                  "-f",
+                  name,
+                  "-r",
+                  rocblas_precision_string<Ti>,
+                  "-n",
+                  n,
+                  "--incx",
+                  incx,
+                  "--stride_x",
+                  stridex,
+                  "--batch",
+                  batch_count);
+    }
+    else
+    {
+        log_bench(handle,
+                  "./rocblas-bench",
+                  "-f",
+                  name,
+                  "-r",
+                  rocblas_precision_string<Ti>,
+                  "-n",
+                  n,
+                  "--incx",
+                  incx);
+    }
+}
+
+template <bool ISBATCHED, typename Ti>
+void reduction_log_bench(rocblas_handle   handle,
+                         rocblas_int      n,
+                         const Ti* const* x,
+                         rocblas_int      incx,
+                         rocblas_stride   stridex,
+                         rocblas_int      batch_count,
+                         const char*      name)
+{
+    log_bench(handle,
+              "./rocblas-bench",
+              "-f",
+              name,
+              "-r",
+              rocblas_precision_string<Ti>,
+              "-n",
+              n,
+              "--incx",
+              incx,
+              "--batch",
+              batch_count);
+}
```

Comment:
Alternatively, we could add _another_ template parameter to differentiate between batched & strided_batched.